### ISSUE TITLE
New component: Choose or create related object

### ISSUE THREAD

**Issue by mihow** [RolnickLab/ami-platform#258]
One general UI component & workflow we need to implement is a "Choose Existing or Create New" form for related types. For example Deployments have a Device (camera type), a Research Site and a Data Source associated with them. These are all related types with their own fields (camera model name, resolution, lighting type, etc). You should be able to select an existing Device to associate with the Deployment or create a new Device. Something to think about! I am adding & updating these related types in the Django admin for now.

I am fine to limit this to "Choose existing" only for now. And force users to create the related types ahead of time in a different form.

![image](https://github.com/RolnickLab/ami-platform/assets/158175/c7ad241e-8bf9-43fe-9000-dc625923005c)
![image](https://github.com/RolnickLab/ami-platform/assets/158175/b3c72151-1934-4ad9-a45a-4cb1ae645dd2)

**Comment by annavik** [comment]
Yes!! This will really improve the experience of creating and editing deployments.
